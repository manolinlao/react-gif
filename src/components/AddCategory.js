import React, { useState } from 'react'

export const AddCategory = ( { setCategories } ) => {

  const [inputValue,setInputValue] = useState('');

  const handleAdd = () => {
    if(inputValue.trim().length>2){
      setCategories( cats => [...cats,inputValue] );
      setInputValue('');
    }   
  }

  const handleInputChange = ( e ) => {    
    setInputValue( e.target.value );
  }

  const handleSubmit = ( e ) => {
    e.preventDefault();
    if(inputValue.trim().length>2){
      setCategories( cats => [...cats,inputValue] );
      setInputValue('');
    }    
  }
 
  return (
    <>      
      <form onSubmit={ handleSubmit }>
        <input
          type="text"
          value={ inputValue }
          onChange={ handleInputChange }
        />
      </form>
      <button onClick={ handleAdd }>agrega</button>
    </>
  )
}
