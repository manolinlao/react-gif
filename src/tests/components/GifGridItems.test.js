import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem/>',()=>{

  // Probar que el componentes se muestra correctamente
  test('debe de mostrar <GifGridItem/> correctamente',()=>{        
    const wrapper = shallow(<GifGridItem/>);
    //EvaluaciÃ³n
    expect(wrapper).toMatchSnapshot();
  });

  // Probando que si se recibe un title como prop, este se  muestra correctamente
  test( 'debe tener un parrafo con el title', () => {
    const title = 'un titulo';
    const wrapper = shallow(<GifGridItem title={ title }/>);
    const p = wrapper.find( 'p' );
    expect( p.text().trim()).toBe( title );
  });
  
  // Probando que al recibir el prop url, aparezca la imagen de ese url y el alt correcto
  test( 'debe tener un parrafo con el title', () => {
    const title = 'un titulo';
    const url = "https://localhost/algo.jpg";     
    const wrapper = shallow(<GifGridItem title={ title } url={ url }/>);
    const img = wrapper.find( 'img' );
    expect( img.prop( 'src' ) ).toBe( url );
    expect( img.prop( 'alt' ) ).toBe( title );
  });

});
