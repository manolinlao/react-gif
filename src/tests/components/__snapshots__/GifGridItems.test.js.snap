// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> debe de mostrar <GifGridItem/> correctamente 1`] = `
<div
  className="card"
>
  <img />
  <p />
</div>
`;
